src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs

```diff
@@ -898,6 +878,138 @@ public static int Sum<T>(this ImmutableArray<T> items, Func<T, int> selector)
             return dictionary;
         }
 
+        internal static void AddToMultiValueDictionaryBuilder<K, T>(Dictionary<K, object> accumulator, K key, T item)
+            where K : notnull
+            where T : notnull
+        {
+            if (accumulator.TryGetValue(key, out var existingValueOrArray))
+            {
+                if (existingValueOrArray is ArrayBuilder<T> arrayBuilder)
+                {
+                    // Already a builder in the accumulator, just add to that.
+                }
+                else
+                {
+                    // Just a single value in the accumulator so far.  Convert to using a builder.
+                    arrayBuilder = ArrayBuilder<T>.GetInstance(capacity: 2);
```

Comment:
I am not interested in changing style or or inverting if conditions. Do not see any value in that.

Reply:
the value is not needing a superfluous empty does-nothing body.  That makes reviewing more difficult as there are more codepaths to reason about.

Reply:
it was harder for me to reason about.  i had to actually convince myself of what was going on because of the lack of use of better and clearer patterns. It's fine if that's not a concern of yours, but it's def a concern of mine (especially as someone who has had to work with this code *a lot* recently.